kpops/components/base_components/base_defaults_component.py

```diff
@@ -158,7 +158,7 @@ def defaults_from_yaml(path: Path, key: str) -> dict:
 
     kafka_app_defaults = defaults_from_yaml(Path("/path/to/defaults.yaml"), "kafka-app")
     """
-    content = load_yaml_file(path, substitution=dict(os.environ))
+    content = load_yaml_file(path, substitution=dict(environ))
```

Comment:
is casting to dict still necessary?

Reply:
I would prefer to this because it creates a copy

Reply:
but is it necessary to create a copy in memory?

Reply:
I mean, I understood the code as doing it because of the same thing. os.environ is already a dict too, so I really thought that it was because of this reason. I will remove it then